pkg/steps/template.go

```diff
@@ -510,7 +510,7 @@ func waitForPodCompletion(ctx context.Context, podClient coreclientset.PodInterf
 }
 
 func waitForPodCompletionOrTimeout(ctx context.Context, podClient coreclientset.PodInterface, eventClient coreclientset.EventInterface, name string, completed map[string]time.Time, notifier ContainerNotifier, skipLogs bool) (bool, error) {
-	watcher, err := podClient.Watch(ctx, meta.ListOptions{
+	watcher, err := podClient.Watch(context.Background(), meta.ListOptions{
 		FieldSelector: fields.Set{"metadata.name": name}.AsSelector().String(),
```

Comment:
Makes me jumpy to have _no_ timeout here.  Can we check `ctx.Err()` and pass `ctx` through on `nil` or a 5m timeout on non-`nil`?  Or plumb through TERM and KILL `Context`s, or something?